context/src/bazel_bep/parser.rs

```diff
@@ -0,0 +1,186 @@
+use anyhow::Ok;
+use bazel_bep::types::build_event_stream::{build_event::Payload, file::File::Uri, BuildEvent};
+use serde_json::Deserializer;
+
+#[derive(Debug, Clone, Default)]
+pub struct TestResult {
+    pub cached: bool,
+    pub xml_files: Vec<String>,
+}
+
+const FILE_URI_PREFIX: &str = "file://";
+
+/// Uses proto spec
+/// https://github.com/TylerJang27/bazel-bep/blob/master/proto/build_event_stream.proto based on
+/// https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto
+#[derive(Debug, Clone, Default)]
+pub struct BazelBepParser {
+    bazel_bep_path: String,
+    test_results: Vec<TestResult>,
+    errors: Vec<String>,
+}
+
+impl BazelBepParser {
+    pub fn new(bazel_bep_path: String) -> Self {
+        Self {
+            bazel_bep_path,
+            ..Default::default()
+        }
+    }
+
+    pub fn errors(&self) -> &Vec<String> {
+        &self.errors
+    }
+
+    pub fn uncached_xml_files(&self) -> Vec<String> {
+        self.test_results
+            .iter()
+            .filter_map(|r| {
+                if r.cached {
+                    return None;
+                }
+                Some(r.xml_files.clone())
+            })
+            .flatten()
+            .collect()
+    }
+
+    pub fn print_parsed_results(&self) {
+        if !self.errors.is_empty() {
+            log::warn!("Errors parsing BEP file: {:?}", &self.errors);
+        }
+
+        let (test_count, cached_count) = self.test_results.iter().fold(
+            (0, 0),
+            |(mut test_count, mut cached_count), test_result| {
+                test_count += test_result.xml_files.len();
+                if test_result.cached {
+                    cached_count += test_result.xml_files.len();
+                }
+                (test_count, cached_count)
+            },
+        );
+        log::info!(
+            "Parsed {} ({} cached) test results from BEP file",
+            test_count,
+            cached_count
+        );
+    }
```

Comment:
Optional: I would leave the `log::info` for the CLI create. This method could just return the number of tests and number of cached tests to be used however the caller pleases

Reply:
Ah, fair point. Will handle in follow-up